### DISCUSSION TITLE
Capstone Project, Etherium, Looping process

### DISCUSSION THREAD

**Discussion by tu849035** [teamalgoritma/community#192] [Q&A] [answered]
Salam.

Mau tanya, untuk proses looping saat scraping dari https://www.coingecko.com/en/coins/ethereum/historical_data/?start_date=2020-01-01&end_date=2021-06-30, untuk kolom Market Cap, Volume, Open, & Close, memiliki kode html yg sama. Seperti tampilan di bawah.
Bagaimana caranya untuk membedakan kodenya agar bisa dimasukan dalam proses looping utk scraping?
Mohon solusinya.

Terima Kasih.

![image](https://user-images.githubusercontent.com/107348211/178149268-a725c516-ffce-4173-a518-9679859eaed4.png)

**Comment by yonvisatria20** [comment, marked as answer]
Kalau dari saya, silahkan lihat tag apa yang membungkus tag "td" . Lalu lihat kolom Volume berada di baris  ke berapa di antara tag td (index dimulai dari 0). Lakukan looping pada tag apa yang membungkus si "th" (kolom Date) dan si "td" (Kolom Volume) .

```
temp = [] #initiating a list 
for ... in soup.find_all('...')[1:]: #diisi dengan tag apa yang membungkus th & td
	*sesuaikan_variabelx = ... .find_all('...') #date berada di tag yg mana?
	date = *sesuaikan_variabelx[...].text #Date berada pada index berapa?
	*sesuaikan_variabely= ... .find_all('...') #Volume berada di tag yg mana?
	volume =*sesuaikan_variabely[...].text.strip() #Volume berada di index berapa?

	temp.append((...,...))

temp

```
Silahkan dilengkapi pak, semoga menjawab.

**Reply by tu849035** [reply]
Sudah saya coba lengkapi kodenya tapi tidak berhasil, munculnya seperti di bawah ini.

![image](https://user-images.githubusercontent.com/107348211/178180169-47f4c706-72dd-496b-8d24-8f4d2186c7aa.png)

Apa mungkin saya salah di bagian "for". Karena jujur saya sendiri masih belum paham mengenai sytax "for" ini. karena selama di kelas belum ataupun di video tutorial, tidak ada penjelasan yg mendetail mengenai "for" ini.
Mohon arahannya di mana yg salah?

Terima Kasih.

**Reply by yonvisatria20** [reply]
baik, saya lampirkan saja source code nya pak, silahkan dimodifikasi sesuai kebutuhan

```
for tr in soup.find_all('tr')[1:]:
	ths = tr.find_all('th')
	date = ths[0].text
	tds = tr.find_all('td')
	volume = tds[1].text.strip()

	temp.append((date,volume))
```
Kalau ada yg ingin ditanyakan silahkan pak

**Reply by tu849035** [reply]
Utk scrapingnya sudah berhasil pak.
Tapi saat saya akan rubah tipe datanya, muncul error.

![image](https://user-images.githubusercontent.com/107348211/178184894-045087c1-7fe9-4e54-9471-3a00048eb468.png)

Mohon pencerahannya.

**Reply by triarm** [reply]
Hi @tu849035 ,
Error tersebut muncul karena masih terdapat tanda koma (,) pada kolom Volume. Silahkan untuk menghapus tanda koma tersebut (dapat digunakan methods `str.replace`) kemudian silahkan untuk melakukan converting tipe data kembali.

Semoga membantu :)
